### PR TITLE
feat: ShouldBindBodyWith shortcut and change doc

### PR DESCRIPTION
**Pull request by RedCrazyGhost**
Resolves https://github.com/gin-gonic/gin/issues/3823

Based on this https://github.com/gin-gonic/gin/issues/3823 proposal, I code about shortcuts

Ran into an issue like https://github.com/gin-gonic/gin/issues/3862 during testing

I hope to get some help.

This is my first time submitting code in open source。

### CODE REVIEW

**Comment by RedCrazyGhost** [comment]
I looked it up, and Red Hat says that YAML is a superset of JSON, and all the data can be parsed in JSON format.

I set out to fix the test case.

**Comment by RedCrazyGhost** [comment]
Hello, please take a look at the PR I submitted and give me some feedback. This PR provides 4 types of shortcuts to ShouldBindBodyWith, which are JSON, YAML, XML, and TOML.

Take a look at this PR, please. @appleboy @thinkerou @manucorporat

**Comment by appleboy** [comment]
I will take a look.

**Comment by appleboy** [comment]
@RedCrazyGhost Please help to fix lint error.

**Comment by RedCrazyGhost** [comment]
> @RedCrazyGhost Please help to fix lint error.

@appleboy  Thanks for your help, I am trying to fix the error in lint, I have committed fix commit.

**Comment by RedCrazyGhost** [comment]
Hello, can you help me start the approval flow? @flc1125

**Comment by flc1125** [comment]
> Hello, can you help me start the approval flow? @flc1125

I'm not a project member and can't help you. Thanks~

**Comment by RedCrazyGhost** [comment]
> > Hello, can you help me start the approval flow? @flc1125
> 
> I'm not a project member and can't help you. Thanks~

Okay, I see. Sorry to bother you.

**Comment by RedCrazyGhost** [comment]
@appleboy Hello, please approve the workflow if you have time

**Review by appleboy — CHANGES_REQUESTED** [review]

File: context_test.go [hunk 1]
```diff
@@ -1975,6 +1975,210 @@ func TestContextShouldBindBodyWith(t *testing.T) {
...
+		w := httptest.NewRecorder()
+		c, _ := CreateTestContext(w)
+
+		c.Request, _ = http.NewRequest("POST", "/", bytes.NewBufferString(tt.body))
+
+		type typeTOML struct {
+			Foo string `toml:"foo" binding:"required"`
+		}
+		objTOML := typeTOML{}
+		if tt.bindingBody != binding.TOML {
```
[the remark this answers is not in the document]
Comment (RedCrazyGhost) [reply context_test.go]: I've changed the code style and clarified the binding type.

Take JSON as an example:
```go
if tt.bindingBody == binding.JSON{
	assert.NoError(t, c.ShouldBindBodyWithJSON(&objJSON))
	assert.Equal(t, typeJSON{"FOO"}, objJSON)
}

if tt.bindingBody == binding.XML {
	assert.Error(t, c.ShouldBindBodyWithJSON(&objJSON))
	assert.Equal(t, typeJSON{}, objJSON)
}

if tt.bindingBody == binding.YAML {
	assert.Error(t, c.ShouldBindBodyWithJSON(&objJSON))
	assert.Equal(t, typeJSON{}, objJSON)
}

if tt.bindingBody == binding.TOML {
	assert.Error(t, c.ShouldBindBodyWithJSON(&objJSON))
	assert.Equal(t, typeJSON{}, objJSON)
}
```

**Comment by RedCrazyGhost** [comment]
Do I need to pay attention to the CodeCov bot detection report? 

Adding new method shortcuts results in a substantial drop in coverage.
@appleboy

**Comment by RedCrazyGhost** [comment]
This codecov detection result is not the same as the result of go test coverage.

This makes me very confused!

I don't know how to solve this patch coverage? @appleboy 

termial use code:
```shell
$ go test -coverprofile=coverage.out ./...
```
```shell
$ go tool cover -html=coverage.out
```

image

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
@RedCrazyGhost I've checked the tests myself, so I won't refer to the data from codecov. :) Thanks.

**Comment by RedCrazyGhost** [comment]
> @RedCrazyGhost I've checked the tests myself, so I won't refer to the data from codecov. :) Thanks.

Thank you for your support throughout.

**Pull request merged**